### ISSUE TITLE
"stack new" can't be pointed at an intranet site

### ISSUE THREAD

**Issue by philipcraig** [commercialhaskell/stack#2804]
### General summary/comments (optional)

"stack new" can't be pointed at an intranet site

### Steps to reproduce

For example:

1. Run command `stack new foo`

### Expected

Some configuration option to get the templates from some intranet site

### Actual

It ends up going to the url https://raw.githubusercontent.com/commercialhaskell/stack-templates/master/new-template.hsfiles which is not on the intranet site.

We've been able to control every other aspect of stack onto our intranet (via the `--stack-setup-yaml` argument, pointing at intranet stuff), but not `stack new`
### Stack version

```
$ stack --version
Version 1.2.0, Git revision 241cd07d576d9c0c0e712e83d947e3dd64541c42 (4054 commits) x86_64 hpack-0.14.0
```

### Method of installation

* Official binary, downloaded from stackage.org or fpcomplete's package repository

**Comment by Blaisorblade** [comment]
Off the top of my head: should that URL become configurable through the description in `--stack-setup-yaml`? I guess that's totally plausible though I lack time to do this any soon...

**Comment by philipcraig** [comment]
Yes, that would work. Or stack new could itself take a template location. Or it could be in the global stack.yaml file

**Comment by jpnp** [comment]
Related to #1980.  There seem a few ideas in this area, such as #2231.  I'd find any of them useful if implemented.

**Comment by alexeyzab** [comment]
Hi there! Is this issue still relevant? If so, would any of the maintainers be interested in providing some mentorship/general help on resolving this so that the newcomers and first-time contributors would have an easier time?

I would like to add this as one of the issues for the upcoming [Haskell Weekly's](https://github.com/haskellweekly/haskellweekly.github.io) Call for Participation section. See discussion in [#75](https://github.com/haskellweekly/haskellweekly.github.io/issues/75).

These are the guidelines we'd like to stick to in the future:
* Ensure that your project has at least one open-source licence. (we've decided to not define the term "open-source" and it is left to your own interpretation).
* Ensure that the issue tracker for your project is publicly accessible.
* Create a new issue in your issue tracker and clearly describe the task. Also mention the difficulty level (easy/medium/hard/tedious), either as a tag/label or somewhere in the title/description.
* If you have specific requirements for contributors (e.g., copyright waiver), it must be mentioned in the description of the task (preferably with a link to CONTRIBUTING.md).

Thank you!

**Comment by AleXoundOS** [comment]
@alexeyzab, yes the issue is relevant. The core problem has much wider scope: **stack** cannot work **offline**. Personally, I've developed a [script](https://github.com/AleXoundOS/haskell-stack-mirror-script) that downloads everything stack needs. Although in order to use the downloaded stuff, you still need manual operations. Some of them are required only once to configure the offline mirror and others every time you run "stack setup", "stack new", which becomes complicated for newcomers.

**Comment by juanpaucar** [comment]
@Blaisorblade I think I could work on this. Is it okay if I take this issue?
It'd be also very helpful if you could point me where the parsing is being done after passed `--stack-setup-yaml`

**Comment by dbaynard** [comment]
I believe this has been fixed — see #4103 — so I'm closing. See [GUIDE.md](https://github.com/commercialhaskell/stack/blob/87b0e7cda7ee3047c4357f0088e206d484c030b1/doc/GUIDE.md#templates) for the details. To use the new version, you'll need to `stack upgrade --git`.

Perhaps, @juanpaucar, you could test to confirm it has been fixed? And then we can find a new issue for which you can implement a feature!
